versioned/persist/adapter/src/test/java/org/projectnessie/versioned/persist/adapter/spi/TestTryLoopState.java

```diff
@@ -31,12 +32,62 @@
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.ArgumentMatcher;
+import org.mockito.InOrder;
 import org.projectnessie.versioned.ReferenceRetryFailureException;
 import org.projectnessie.versioned.persist.adapter.DatabaseAdapterConfig;
 import org.projectnessie.versioned.persist.adapter.spi.TryLoopState.MonotonicClock;
 
 class TestTryLoopState {
 
+  @ParameterizedTest
+  @ValueSource(longs = {1, 5, 50, 100, 200})
+  void doesNotSleepLongerThanMax(long maxSleep) throws Exception {
+    int retries = 50;
+    Long[] times = new Long[retries];
+    Arrays.fill(times, 0L);
+    MonotonicClock clock = mockedClock(0L, times);
+
+    long lower = 1;
+    long upper = 2;
+    TryLoopState tryLoopState =
+        new TryLoopState(
+            this::retryErrorMessage,
+            mockedConfig(retries, Long.MAX_VALUE, 1, upper, maxSleep),
+            clock,
+            (success, tls) -> {});
+
+    verify(clock, times(1)).currentNanos();
+
+    InOrder inOrderClock = inOrder(clock);
+
+    for (int i = 0; i < retries; i++) {
+      tryLoopState.retry();
+      long finalLower = lower;
+      long finalUpper = upper;
+      ArgumentMatcher<Long> matcher =
+          new ArgumentMatcher<Long>() {
+            @Override
+            public boolean matches(Long l) {
+              return l >= finalLower && l < finalUpper && l <= maxSleep;
```

Comment:
so lower and maxSleep are inclusive while upper is exclusive?
looks odd to my naive eyes, but probably correct :sweat_smile: